grove/connectors/oomnitza/api.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+"""Oomnitza API client.
+
+As Oomnitza SDK does not currently support the Activities API, this client has been created in
+the interim.
+"""
+
+import logging
+from typing import Dict, Optional
+
+import requests
+from grove.exceptions import RequestFailedException
+from grove.types import AuditLogEntries, HTTPResponse
+
+API_BASE_URI = "https://{identity}.oomnitza.com"
+API_PAGE_SIZE = 200
+
+
+class Client:
+    def __init__(
+        self,
+        identity: Optional[str] = None,
+        token: Optional[str] = None,
+    ):
+        """Setup a new client.
+
+        :param identity: The name of the Oomnitza organisation.
+        :param token: The Oomnitza API token.
+        """
+        self.logger = logging.getLogger(__name__)
+        self.headers = {
+            "content-type": "application/json",
+            "Authorization2": f"{token}",
```

Comment:
Can token really be None? And do we need the f-string here?

Reply:
Good catch. This may not have been caught by the linter due to `Optional`. That said, it looks like there's a dependency issue failing the actions runs, so we'll need to pin that dependency to allow CI to complete to validate :)

Reply:
Taking a look at the type information for `requests`, it appears that the [supported types](https://github.com/python/typeshed/blob/b69b17c3d8fd5b1f0cc8209b2f15e6b4b687a2ee/stubs/requests/requests/api.pyi#L8) for `_HeadersMapping` is `str | bytes` for the value. As a result, if we used `token` without the f-string (which is an `Optional[str]`), we would violate that type.

That said, if `token` was `None` it would, as written, pass `"None"` as a string which is also not what we're after.

(It's worth noting that `headers` works differently to `params` - [where `_ParamsMappingValueType` allows `None`](https://github.com/python/typeshed/blob/b69b17c3d8fd5b1f0cc8209b2f15e6b4b687a2ee/stubs/requests/requests/sessions.pyi#LL96C1-L96C24) and [removes the key from the map automatically](https://github.com/psf/requests/issues/2651). As above, this is not the case with `headers`, so we need to handle it.)

I would suggest shifting this slightly to only add `Authorization2` to the `self.headers` if there is a value set for `token`, otherwise omit it. This way, if the remote API supports unauthenticated query, the `Authorization2` header is never included if no value is set in the configuration :) 

Although not necessarily the best way to achieve this, something like the following should do what we'd be after:

```suggestion
        }

        if token:
            self.headers["Authorization2"] = token

```

Reply:
okay, make sense. Thank you!